sdk/tests/DeserializationTests.cc

```diff
@@ -0,0 +1,254 @@
+
+  auto testProtoExchangeRate = std::make_unique<proto::ExchangeRate>();
+  testProtoExchangeRate->set_centequiv(testCents * testCents);
+  testProtoExchangeRate->set_hbarequiv(testHbar * testHbar);
+
+  proto::ExchangeRateSet testProtoExchangeRateSet;
+  testProtoExchangeRateSet.set_allocated_nextrate(testProtoExchangeRate.release());
+
+  // When
+  ExchangeRateSet exchangeRateSet = ExchangeRateSet::fromProtobuf(testProtoExchangeRateSet);
+  
+  // Then
+  EXPECT_FALSE(exchangeRateSet.getCurrentExchangeRate().has_value());
+  EXPECT_TRUE(exchangeRateSet.getNextExchangeRate().has_value());
+  EXPECT_EQ(exchangeRateSet.getNextExchangeRate().value().getCurrentExchangeRate(), testCents * testCents / testHbar * testHbar);
+}
+
+TEST_F(DeserializationTests, DeserializeTransactionReceiptFromProtobufTest)
+{
+  // Given
+  const proto::ResponseCodeEnum testResponseStatus = proto::ResponseCodeEnum::SUCCESS;
+  proto::AccountID *testProtoAccountId = getTestAccountId().toProtobuf().release();
+  proto::TransactionReceipt testProtoTxReceipt;
+  testProtoTxReceipt.set_status(testResponseStatus);
+  testProtoTxReceipt.set_allocated_accountid(testProtoAccountId);
+
+  const int32_t value = 6;
+  const int32_t secs = 100;
+
+  proto::ExchangeRateSet *protoExRateSet = testProtoTxReceipt.mutable_exchangerate();
+  protoExRateSet->mutable_currentrate()->set_hbarequiv(value);
+  protoExRateSet->mutable_currentrate()->set_centequiv(value);
+  protoExRateSet->mutable_currentrate()->mutable_expirationtime()->set_seconds(secs);
+  protoExRateSet->mutable_nextrate()->set_hbarequiv(value);
+  protoExRateSet->mutable_nextrate()->set_centequiv(value);
+  protoExRateSet->mutable_nextrate()->mutable_expirationtime()->set_seconds(secs);
+
+  // When
+  TransactionReceipt txRx = TransactionReceipt::fromProtobuf(testProtoTxReceipt);
+  
+  // Then
+  EXPECT_EQ(txRx.getStatus(), Status::SUCCESS);
+  EXPECT_EQ(txRx.getAccountId(), getTestAccountId());
+  EXPECT_TRUE(txRx.getExchangeRates().has_value());
+  EXPECT_TRUE(txRx.getExchangeRates()->getCurrentExchangeRate().has_value());
+  EXPECT_EQ(txRx.getExchangeRates()->getCurrentExchangeRate()->getCurrentExchangeRate(), value / value);
+  EXPECT_TRUE(txRx.getExchangeRates()->getCurrentExchangeRate()->getExpirationTime().has_value());
+  EXPECT_EQ(txRx.getExchangeRates()->getCurrentExchangeRate()->getExpirationTime(),
+            std::chrono::system_clock::time_point(std::chrono::seconds(secs)));
+  EXPECT_TRUE(txRx.getExchangeRates()->getNextExchangeRate().has_value());
+  EXPECT_EQ(txRx.getExchangeRates()->getNextExchangeRate()->getCurrentExchangeRate(), value / value);
+  EXPECT_TRUE(txRx.getExchangeRates()->getNextExchangeRate()->getExpirationTime().has_value());
+  EXPECT_EQ(txRx.getExchangeRates()->getNextExchangeRate()->getExpirationTime(),
+            std::chrono::system_clock::time_point(std::chrono::seconds(secs)));
+}
+
+TEST_F(DeserializationTests, DeserializeTransactionRecordFromProtobufTest)
```

Comment:
To write this test I used test [ProtobufTransactionRecord](https://github.com/hashgraph/hedera-sdk-cpp/blob/c54139e8bc8f507e1f08d052b15eec3d188a0bd0/sdk/tests/TransactionRecordTest.cc#L34) from [TransactionRecordTest.cc](https://github.com/hashgraph/hedera-sdk-cpp/blob/main/sdk/tests/TransactionRecordTest.cc).

Reply:
Remove test `DeserializeTransactionRecordFromProtobufTest ` from `DeserializationTests.cc` in commit https://github.com/hashgraph/hedera-sdk-cpp/pull/176/commits/679c14b787e9d21e506dfa2add686593a61f7863.